### ISSUE TITLE
ejabberd: Enable XEPs

### ISSUE THREAD

**Issue by jvalleroy** [freedombox/Plinth#703]
According to https://github.com/iNPUTmice/ComplianceTester, the following XEPs are not enabled in our ejabberd configuration, but should be:
- XEP-0237: Roster Versioning (mod_roster)
- XEP-0313: Message Archive Management (mod_mam)
- XEP-0363: HTTP File Upload (mod_http_upload)

(Also XEP-0357: Push Notifications, but that is not officially supported by ejabberd yet.)

**Comment by jvalleroy** [comment]
Ejabberd bug report: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=851212

mod_roster can be enabled by default in the Debian package. For the other two, we will need to understand the requirements and security/privacy implications.

**Comment by mray** [comment]
Apart from finding sensible defaults: wouldn't it be worth exposing enabed XEPs to Plinth?

**Comment by JoKeyser** [comment]
I agree with both of you; I think we should find sensible, cautious defaults, but ultimately leave the choice to the user, along with a warning about the implications and point to resources if they need more info.

So to get ready for implementation, I propose
1. Expose checkboxes to enable/disable XEPs on ejabberd configuration page, along with explanations. 
2. Potentially monitor the size of the mnesia database used by mod_mam.
3. Potentially monitor the used storage for mod_http_upload.

There is a nice mapping of ejabberd configs to XEPs at https://gitlab.com/hanno/ejabberd-config.

**Comment by JoKeyser** [comment]
@jvalleroy, did you manage to test vagrant box's ejabberd with https://github.com/iNPUTmice/ComplianceTester? If so, how? I've somewhat blindly tried by first `apt-get install default-jdk` and then tried to run it within the vagrant using localhost... but it seems to get stuck, without a clear error message (at least for me):

```
root@freedombox:/vagrant # java -jar ComplianceTester-0.1.jar testuser@localhost testpassword
Use compliance suite 'Advanced Server Core Compliance Suite' to test localhost

Timeout while waiting on advertised authentication mechanisms.

Use compliance suite 'Advanced Server IM Compliance Suite' to test localhost

Use compliance suite 'Advanced Server Mobile Compliance Suite' to test localhost

Use compliance suite 'Conversations Compliance Suite' to test localhost
```

**Comment by JoKeyser** [comment]
I've got the ComplianceTester working now with the vagrant image, making further development easier. It's currently a bit annoying since you need to create a new certificate for ejabberd, because java seems to insist that the hostname is in the cert's common name, which currently isn't the case in the vagrant image.
These steps assume you're in the Plinth root folder (/vagrant in the virtual image), and run the ComplianceTester from the host machine (another way may be to run it from within vagrant, but then you need to first install java).

First, you need to forward ejabberd's c2s port to the host machine, by adding this line into `Vagrantfile`:  
`  config.vm.network "forwarded_port", guest: 5222, host: 5222`

FIXME: Edit in the rest.

**Comment by JoKeyser** [comment]
On to the third, XEP-0363: HTTP File Upload.
This is the most complicated one; it needs more configuration, including the firewall. That means more defaults - let's discuss them here, and document the decision process. I think we also need `mod_http_upload_quota`, to regulate how much disk space a user can use. (I'm afraid the alternative is that it keeps filling until the disk is full.)

The core assumption behind my suggestions below is that ejabberd serves only a few users, maybe 10-15 (not hundreds). I think these few users should not be hampered by too little upload space (>1 GiB, if possible). On the other hand, we're targeting small machines, and ejabberd should not unduly interfere with other services (<50% of all). I'm assuming here at least 32 GiB in the box, so setting a hard limit of 1.1 GiB per user would result in just above 50% of that machine.

Maybe some settings need to be exposed in Plinth?
Maybe reporting them to the admin is suffcient?
Maybe some default must be adapted to the actual machine that's configured?


Below are the interesting settings I think we should discuss the defaults of.
For those I don't mention, I think ejabberd's defaults clearly make sense.
See https://docs.ejabberd.im/admin/configuration/#mod-http-upload for the details (those of `mod_http_upload_quota` is just below).

Drafted defaults for `mod_http_upload`:
* `max_size:` Maximal size of a single file. Ejabberd default is 100 MiB. Seems fine?
* `docroot:` Where to place the files. Ejabberd default is on root partition. Give user the choice?

Drafted defaults for `mod_http_upload_quota`:
* `max_days:` Number of days until files are deleted. Ejabberd default is infinity. I'd rather like 365?
* `access_hard_quota:` Hard limit per user. If exceeded, oldest files get erased. Ejabberd default is unspecified. I suggest 1.1 GiB.
* `access_soft_quota:` Soft limit per user. Unclear to me what happens if exceeded. Ejabberd default  unspecified. I suggest 1.0 GiB.

**Comment by JoKeyser** [comment]
A rather random item for context: The maximal upload size of a single file in Matrix Synapse is also 100MiB per default. So really seems fine.
